### ISSUE TITLE
[scylla deployment failure] Scylla Manager Agent fails to connect to Scylla API

### ISSUE THREAD

**Issue by vponomaryov** [scylladb/scylla-operator#454]
**Describe the bug**
Installing scylla-operator , scylla-manager and scylla using helm charts causes networking problems in scylla [1] and scylla-manager [2].

More details about helm charts configs and resources are in attachment.

[1] Scylla server (the one created by user)
``` {"L":"INFO","T":"2021-03-02T09:43:46.215Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 0.0.0.0:10000: connect: connection refused","_trace_id":"O3gymFQ9Twa6fwYX5K7Gcw"}
{"L":"ERROR","T":"2021-03-02T09:43:48.215Z","M":"Bye","error":"server init: no connection to Scylla API, make sure that Scylla server is running and api_address and api_port are set correctly in config file [/etc/scylla-manager-agent/scylla-manager-agent.yaml /mnt/scylla-agent-config/scylla-manager-agent.yaml]: giving up after 60 attempts: dial tcp 0.0.0.0:10000: connect: connection refused","_trace_id":"O3gymFQ9Twa6fwYX5K7Gcw","errorStack":"main.glob..func3\n\tgithub.com/scylladb/scylla-manager@/pkg/cmd/agent/root.go:62\ngithub.com/spf13/cobra.(*Command).execute\n\tgithub.com/spf13/cobra@v0.0.5/command.go:826\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\tgithub.com/spf13/cobra@v0.0.5/command.go:914\ngithub.com/spf13/cobra.(*Command).Execute\n\tgithub.com/spf13/cobra@v0.0.5/command.go:864\nmain.main\n\tgithub.com/scylladb/scylla-manager@/pkg/cmd/agent/main.go:15\nruntime.main\n\truntime/proc.go:203\nruntime.goexit\n\truntime/asm_amd64.s:1357\n"}

STARTUP ERROR: server init: no connection to Scylla API, make sure that Scylla server is running and api_address and api_port are set correctly in config file [/etc/scylla-manager-agent/scylla-manager-agent.yaml /mnt/scylla-agent-config/scylla-manager-agent.yaml]: giving up after 60 attempts: dial tcp 0.0.0.0:10000: connect: connection refused
```
[2] Scylla Manager
```
{"L":"INFO","T":"2021-03-02T08:37:16.070Z","M":"Scylla Manager Server","version":"2.2.1-0.20201209.47b1a38f","pid":1,"_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:17.006Z","M":"Using config","config":{"HTTP":":5080","HTTPS":"","TLSCertFile":"/var/lib/scylla-manager/scylla_manager.crt","TLSKeyFile":"/var/lib/scylla-manager/scylla_manager.key","TLSCAFile":"","Prometheus":":5090","Debug":"127.0.0.1:5112","SwaggerUIPath":"/var/lib/scylla-manager/swagger-ui","Logger":{"Mode":"stderr","Level":"info","Development":false},"Database":{"Hosts":["scylla-manager-manager-dc-manager-rack-0.scylla-manager-system.svc"],"SSL":false,"User":"","Password":"","LocalDC":"","Keyspace":"scylla_manager","MigrateDir":"/etc/scylla-manager/cql","MigrateTimeout":30000000000,"MigrateMaxWaitSchemaAgreement":300000000000,"ReplicationFactor":1,"Timeout":600000000,"TokenAware":true},"SSL":{"CertFile":"","Validate":true,"UserCertFile":"","UserKeyFile":""},"Healthcheck":{"Timeout":250000000,"SSLTimeout":750000000,"DynamicTimeout":{"Enabled":true,"Probes":200,"MaxTimeout":30000000000,"StdDevMultiplier":5},"NodeInfoTTL":300000000000},"Backup":{"DiskSpaceFreeMinPercent":10,"LongPollingTimeoutSeconds":10,"AgeMax":43200000000000},"Repair":{"PollInterval":50000000,"LongPollingTimeoutSeconds":10,"AgeMax":0,"GracefulStopTimeout":30000000000,"ForceRepairType":"auto","Murmur3PartitionerIgnoreMSBBits":12}},"config_files":["/mnt/etc/scylla-manager/scylla-manager.yaml"],"_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:17.007Z","M":"Checking database connectivity...","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:18.007Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: i/o timeout","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:20.007Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: i/o timeout","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:22.007Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: i/o timeout","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:24.007Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: i/o timeout","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:26.007Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: i/o timeout","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:27.011Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:29.010Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:31.010Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:33.010Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:35.011Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:37.010Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:39.011Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:41.011Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:43.011Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:45.011Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:47.011Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:49.012Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:51.012Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:53.013Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:55.012Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:57.012Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:37:59.013Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:01.013Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:03.013Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:05.013Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:07.013Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:09.013Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:11.013Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:13.014Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:15.014Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:17.014Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:19.014Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:21.014Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:23.014Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:25.015Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:27.015Z","N":"wait","M":"Waiting for network connection","sleep":"2s","error":"dial tcp 10.3.250.43:9042: connect: connection refused","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:29.060Z","M":"Creating keyspace","keyspace":"scylla_manager","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:29.134Z","M":"Keyspace created","keyspace":"scylla_manager","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:29.134Z","M":"Migrating schema","keyspace":"scylla_manager","dir":"/etc/scylla-manager/cql","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:29.339Z","M":"Running migration","migration":"001-init_scheduler.cql"}
{"L":"INFO","T":"2021-03-02T08:38:29.816Z","M":"Running migration","migration":"002-init_repair.cql"}
{"L":"INFO","T":"2021-03-02T08:38:30.588Z","M":"Running migration","migration":"003-init_cluster.cql"}
{"L":"INFO","T":"2021-03-02T08:38:30.778Z","M":"Running migration","migration":"004-repair_run_progress_segment_error_start_tokens.cql"}
{"L":"INFO","T":"2021-03-02T08:38:30.955Z","M":"Running migration","migration":"005-drop_repair_config.cql"}
{"L":"INFO","T":"2021-03-02T08:38:31.243Z","M":"Running migration","migration":"006-ssh_user_per_cluster.cql"}
{"L":"INFO","T":"2021-03-02T08:38:31.398Z","M":"Running migration","migration":"007-task_properties_as_blob.cql"}
{"L":"INFO","T":"2021-03-02T08:38:31.779Z","M":"Running migration","migration":"008-drop_repair_unit.cql"}
{"L":"INFO","T":"2021-03-02T08:38:32.972Z","M":"Running migration","migration":"009-scheduler_remove_task_metadata.cql"}
{"L":"INFO","T":"2021-03-02T08:38:33.153Z","M":"Running migration","migration":"010-cluster_drop_shard_count.cql"}
{"L":"INFO","T":"2021-03-02T08:38:33.336Z","M":"Running migration","migration":"011-cluster_add_host.cql"}
{"L":"INFO","T":"2021-03-02T08:38:33.741Z","M":"Running migration","migration":"013-update_ttl.cql"}
{"L":"INFO","T":"2021-03-02T08:38:34.319Z","M":"Running migration","migration":"014-scheduler_rename_interval_days_to_interval_seconds.cql"}
{"L":"INFO","T":"2021-03-02T08:38:34.620Z","M":"Running migration","migration":"015-cluster_add_known_hosts.cql"}
{"L":"INFO","T":"2021-03-02T08:38:34.817Z","M":"Running migration","migration":"016-cluster_drop_host.cql"}
{"L":"INFO","T":"2021-03-02T08:38:35.032Z","M":"Running migration","migration":"018-empty.cql"}
{"L":"INFO","T":"2021-03-02T08:38:35.047Z","M":"Running migration","migration":"019-repair_segment_error_pos.cql"}
{"L":"INFO","T":"2021-03-02T08:38:35.233Z","M":"Running migration","migration":"v2.0.0.cql"}
{"L":"INFO","T":"2021-03-02T08:38:37.775Z","M":"Running migration","migration":"v2.1.0.cql"}
{"L":"INFO","T":"2021-03-02T08:38:38.584Z","M":"Running migration","migration":"v2.2.0.cql"}
{"L":"INFO","T":"2021-03-02T08:38:39.862Z","M":"Post migration","action":"fix TTL in scheduler_task table"}
{"L":"INFO","T":"2021-03-02T08:38:40.048Z","M":"Schema up to date","keyspace":"scylla_manager","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:40.071Z","N":"scheduler","M":"Loading tasks from database","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:40.073Z","N":"scheduler","M":"All tasks scheduled","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:40.073Z","M":"Starting HTTP server","address":":5080","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:40.073Z","M":"Starting Prometheus server","address":":5090","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:40.073Z","M":"Starting debug server","address":"127.0.0.1:5112","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:40.073Z","M":"Service started","_trace_id":"LTMEA9_iQ7OqdnmaHCuuCw"}
{"L":"INFO","T":"2021-03-02T08:38:41.778Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"GET","uri":"/api/v1/version","status":200,"bytes":40,"duration":"0ms","_trace_id":"gNyWqmpqQz2vdlI5mcyxnA"}
{"L":"INFO","T":"2021-03-02T08:39:04.676Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"GET","uri":"/api/v1/clusters","status":200,"bytes":3,"duration":"2ms","_trace_id":"gEQ3_YasTEC8gGjtdvWMLQ"}
{"L":"INFO","T":"2021-03-02T08:39:05.278Z","N":"cluster","M":"Adding new cluster","cluster_id":"c0a921bd-1ec8-4827-806f-0872e6cabb3f","_trace_id":"mARLANYHT3GDow8ku7Uc2A"}
{"L":"INFO","T":"2021-03-02T08:39:05.292Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"POST","uri":"/api/v1/clusters","status":500,"bytes":263,"duration":"14ms","error":"create cluster: status: giving up after 1 attempts: dial tcp: lookup sct-cluster-client.scylla.svc.cluster.local on 10.3.240.10:53: no such host - make sure the IP is correct and access to port 10001 is unblocked","_trace_id":"mARLANYHT3GDow8ku7Uc2A"}
{"L":"INFO","T":"2021-03-02T08:39:06.299Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"GET","uri":"/api/v1/clusters","status":200,"bytes":3,"duration":"1ms","_trace_id":"avKm-ETZRl6BGB0AEbkXEg"}
{"L":"INFO","T":"2021-03-02T08:39:06.300Z","N":"cluster","M":"Adding new cluster","cluster_id":"d8267bc9-212f-4b6a-93c0-96740a81e3de","_trace_id":"_GTG07Y1Rb2kOUZ12UMK6A"}
{"L":"INFO","T":"2021-03-02T08:39:06.305Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"POST","uri":"/api/v1/clusters","status":500,"bytes":263,"duration":"5ms","error":"create cluster: status: giving up after 1 attempts: dial tcp: lookup sct-cluster-client.scylla.svc.cluster.local on 10.3.240.10:53: no such host - make sure the IP is correct and access to port 10001 is unblocked","_trace_id":"_GTG07Y1Rb2kOUZ12UMK6A"}
{"L":"INFO","T":"2021-03-02T08:39:07.312Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"GET","uri":"/api/v1/clusters","status":200,"bytes":3,"duration":"1ms","_trace_id":"rq-VZxCRSgaP9iDd7jqvJA"}
{"L":"INFO","T":"2021-03-02T08:39:07.312Z","N":"cluster","M":"Adding new cluster","cluster_id":"cf4a566b-9d1c-47b9-9973-0e443b39c622","_trace_id":"_g-vNzQPRcaJwXauN1giVA"}
{"L":"INFO","T":"2021-03-02T08:39:07.319Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"POST","uri":"/api/v1/clusters","status":500,"bytes":263,"duration":"6ms","error":"create cluster: status: giving up after 1 attempts: dial tcp: lookup sct-cluster-client.scylla.svc.cluster.local on 10.3.240.10:53: no such host - make sure the IP is correct and access to port 10001 is unblocked","_trace_id":"_g-vNzQPRcaJwXauN1giVA"}
{"L":"INFO","T":"2021-03-02T08:39:08.604Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"GET","uri":"/api/v1/clusters","status":200,"bytes":3,"duration":"1ms","_trace_id":"g74hs8AHRwuP_7IkddLNjA"}
{"L":"INFO","T":"2021-03-02T08:39:08.605Z","N":"cluster","M":"Adding new cluster","cluster_id":"71aa86f6-e352-46be-97e5-ea3973a673c2","_trace_id":"1W8ERVVOSMC0QJcK31GYIQ"}
{"L":"INFO","T":"2021-03-02T08:39:08.611Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"POST","uri":"/api/v1/clusters","status":500,"bytes":263,"duration":"6ms","error":"create cluster: status: giving up after 1 attempts: dial tcp: lookup sct-cluster-client.scylla.svc.cluster.local on 10.3.240.10:53: no such host - make sure the IP is correct and access to port 10001 is unblocked","_trace_id":"1W8ERVVOSMC0QJcK31GYIQ"}
{"L":"INFO","T":"2021-03-02T08:39:11.177Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"GET","uri":"/api/v1/clusters","status":200,"bytes":3,"duration":"1ms","_trace_id":"3uHWRtJ0TlSiaJDdZy6NbQ"}
{"L":"INFO","T":"2021-03-02T08:39:11.178Z","N":"cluster","M":"Adding new cluster","cluster_id":"ec8089cf-8761-46b0-b982-b2cebfc0c5fc","_trace_id":"x86OIuwgTdK3m_-GuMRwBw"}
{"L":"INFO","T":"2021-03-02T08:39:11.195Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"POST","uri":"/api/v1/clusters","status":500,"bytes":263,"duration":"16ms","error":"create cluster: status: giving up after 1 attempts: dial tcp: lookup sct-cluster-client.scylla.svc.cluster.local on 10.3.240.10:53: no such host - make sure the IP is correct and access to port 10001 is unblocked","_trace_id":"x86OIuwgTdK3m_-GuMRwBw"}
{"L":"INFO","T":"2021-03-02T08:39:16.320Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"GET","uri":"/api/v1/clusters","status":200,"bytes":3,"duration":"1ms","_trace_id":"KYt5cKxpTvSscYnRT1w9tA"}
{"L":"INFO","T":"2021-03-02T08:39:16.321Z","N":"cluster","M":"Adding new cluster","cluster_id":"63fac19d-f7c4-4fc0-9d24-a32461d019af","_trace_id":"fgkdit8jSfKVnkVwr9c5BA"}
{"L":"INFO","T":"2021-03-02T08:39:16.336Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"POST","uri":"/api/v1/clusters","status":500,"bytes":263,"duration":"15ms","error":"create cluster: status: giving up after 1 attempts: dial tcp: lookup sct-cluster-client.scylla.svc.cluster.local on 10.3.240.10:53: no such host - make sure the IP is correct and access to port 10001 is unblocked","_trace_id":"fgkdit8jSfKVnkVwr9c5BA"}
{"L":"INFO","T":"2021-03-02T08:39:26.583Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"GET","uri":"/api/v1/clusters","status":200,"bytes":3,"duration":"1ms","_trace_id":"FyNYF43zTWO6uAjzvhba6w"}
{"L":"INFO","T":"2021-03-02T08:39:26.583Z","N":"cluster","M":"Adding new cluster","cluster_id":"a166f6c3-7c86-437a-a6da-942a1d020d6a","_trace_id":"Y4mQfvGnSs2hvPx9kniyQg"}
{"L":"INFO","T":"2021-03-02T08:39:26.604Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"POST","uri":"/api/v1/clusters","status":500,"bytes":263,"duration":"20ms","error":"create cluster: status: giving up after 1 attempts: dial tcp: lookup sct-cluster-client.scylla.svc.cluster.local on 10.3.240.10:53: no such host - make sure the IP is correct and access to port 10001 is unblocked","_trace_id":"Y4mQfvGnSs2hvPx9kniyQg"}
{"L":"INFO","T":"2021-03-02T08:39:47.090Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"GET","uri":"/api/v1/clusters","status":200,"bytes":3,"duration":"1ms","_trace_id":"nhbhIgEJTKWWYMSNYTbPog"}
{"L":"INFO","T":"2021-03-02T08:39:47.090Z","N":"cluster","M":"Adding new cluster","cluster_id":"8d072b63-a0a3-48c9-99ee-ce5b570f0623","_trace_id":"o0fYxkwFQY27QBHunIAe1w"}
{"L":"INFO","T":"2021-03-02T08:39:47.107Z","N":"http","M":"HTTP","from":"10.0.4.20:60594","method":"POST","uri":"/api/v1/clusters","status":500,"bytes":263,"duration":"16ms","error":"create cluster: status: giving up after 1 attempts: dial tcp: lookup sct-cluster-client.scylla.svc.cluster.local on 10.3.240.10:53: no such host - make sure the IP is correct and access to port 10001 is unblocked","_trace_id":"o0fYxkwFQY27QBHunIAe1w"}
```

**To Reproduce**
1. Install scylla-operator-v1.1.0-rc.2-1-g9b93ca0 , scylla-manager-v1.1.0-rc.2-1-g9b93ca0 and scylla-v1.1.0-rc.2-1-g9b93ca0
2. See error

**Expected behavior**
Expected successful creation of Scylla server without any DNS and networking issues

**Config Files**
Helm configuration and k8s resources data are added to the "helm-and-k8s-data.tar.gz" archive.

**Logs**
Please, provide `kubectl get events`, `kubectl logs -n scylla pod-name` and other relevant information.

**Environment:**
 - Platform: GKE
 - Kubernetes version: 1.17.15-gke.800
 - Scylla version: 4.3.0
 - Scylla Manager version: 2.2.1
 - Scylla Manager agent version: 2.2.1
 - Scylla-operator version: e.g.: v1.1.0-rc.2-1-g9b93ca0

**Additional context**
- Installing approximately the same config using sample files doesn't have such problem.
- Installing scylla server using sample files when scylla-operator and scylla-manager are installed using helm charts fails the same way.
- 
[helm-and-k8s-data.tar.gz](https://github.com/scylladb/scylla-operator/files/6067882/helm-and-k8s-data.tar.gz)
Problem gets reproduced constantly in 100% cases.

**Comment by zimnx** [comment]
Known issue of Scylla 4.3.0, try with 4.2.x https://github.com/scylladb/scylla/issues/8032

**Comment by vponomaryov** [comment]
@zimnx Using scylla 4.2.0 problem goes away. Thanks

**Comment by vponomaryov** [comment]
@zimnx I think that "scylla-operator" logic can be improved.
"scylla" container must not have been in "Ready" state not really being operational due to the failed "scylla setup" step.
It must have been in "Error" state and appropriate "event" recorded.

**Comment by zimnx** [comment]
readiness probe for Scylla container requires node to be in UN state and have CQL port open, so i don't see what can be improved here. Are you sure you were looking at the right pod?

**Comment by vponomaryov** [comment]
> readiness probe for Scylla container requires node to be in UN state and have CQL port open, so i don't see what can be improved here. Are you sure you were looking at the right pod?

I am referring to the "scylla" container where "scylla_io_setup" step failed. In attached logs it is "k8s/scylla--scylla.log" file.
Second container in it's pod fails to start up, it is "manager-agent". And it fails just because the first one hadn't started up correctly having "Running" state.
I haven't checked source code of the scylla-operator, but that Scylla node with failed disk setup could not be in UN state.
So, either UN state is fake or check for it is incorrect.

In summary: "scylla" container is "scylla" pod must be in "Error" state when any of the "setup scylla" steps fails.

**Comment by zimnx** [comment]
Ok in previous comment you reffered to "Ready" state and now to "Running" state. These are two different things, first one is controlled by the Operator, second one by the kublet and container. Ready is fine, it checks whether node is in UN state using API. 

Running is something indeed we should fix, because sidecar process (container) doesn't die when scylla child process dies.

**Comment by vponomaryov** [comment]
@zimnx Sorry for the confusion, my mistake. You are right, it must have been "Running" one in the original message.
